File: boinghop-cs50-problems-2021-spring-sentimental-readability/readability.py
import cs50

text = cs50.get_string("Text:\n")

# intitializing variables for counting
letter = 0 
word = 1
sentence = 0

# iterate over the text
for character in text:
    # if character is a letter
    if character.isalpha():
        # count +1 value
        letter = letter + 1
    # if character is a space
    if character.isspace():
        # count +1 value
        word = word + 1
    # if character is one of these punctuation
    if character == "." or character == "!" or character == "?":
        # count +1 value
        sentence = sentence + 1
        
# get the average over 100 words for letteres and sentences
L = letter / float(word) * 100
S = sentence / float(word) * 100

# formula + round into index
index = round(0.0588 * L - 0.296 * S - 15.8)
if (index < 1):
    print("Before Grade 1")
elif (index > 16):
    print("Grade 16+")
else:
    print(f"Grade {index}")

File: boinghop-cs50-problems-2020-fall-readability/readability.c
#include <stdio.h>
#include <string.h>
#include <cs50.h>
#include <ctype.h>
#include <stdlib.h>
#include <math.h>

int compute_word (string word);

int main(void)

{
    //prompt user for a string of text
    string text = get_string ("Text: ");
    int letter = 0, word = 1, sentence = 0;
    {
        //iterate over the text
        for (int i = 0, n = strlen(text); i < n; i++)
        {
            //increment L for any letter regardless of case
            if ((isupper(text[i])) || (islower(text[i])))
                {
                    letter++;
                }
            //increment W for any space, W is set to 1 to account for the last word
            if ((isspace(text[i])))
                {
                    word++;
                }
            //increment S for these specific punctuation
            if ((text[i] == '.' || text[i] == '!' || text[i] == '?'))
                {
                    sentence++;
                }
        }
    }
        //print letters, words, sentences
        // printf("Letter(s): %i\nWord(s): %i\nSentence(s): %i\n", letter, word, sentence);
       float L = (letter / (float) word) * 100;
       float S = (sentence / (float) word) * 100;
       
       int index = round(0.0588 * L - 0.296 * S - 15.8);
        {
            if (index < 1)
            {
                printf ("Before Grade 1\n");
            }
            else if (index > 16)
            {
                printf("Grade 16+\n");
            }
            else
            {
                printf("Grade %i\n", index);
            }
        }
    }

